projects/igniteui-angular/src/lib/grids/grid/column-group.spec.ts

```diff
@@ -1368,6 +1369,40 @@ describe('IgxGrid - multi-column headers', () => {
             fixture.detectChanges();
         }).not.toThrow();
     }));
+
+    it('Should correctly initialize column group templates.', fakeAsync(() => {
+        const fixture = TestBed.createComponent(NestedColGroupsWithTemplatesGridComponent);
+        fixture.detectChanges();
+        const ci = fixture.componentInstance;
+        const locationColGroup = ci.locationColGroup;
+        const contactInfoColGroup = ci.contactInfoColGroup;
+
+        expect(locationColGroup.headerTemplate).toBeDefined();
+        expect(contactInfoColGroup.headerTemplate).toBeUndefined();
+
+        const headerSpans: DebugElement[] = fixture.debugElement.queryAll(By.css('.col-group-template'));
+        expect(headerSpans.length).toBe(1);
+        expect(headerSpans[0].nativeElement.textContent).toMatch('Column group template');
+    }));
+
+    it('Should correctly change column group templates dynamically.', fakeAsync(() => {
+        const fixture = TestBed.createComponent(NestedColGroupsWithTemplatesGridComponent);
+        fixture.detectChanges();
+        const ci = fixture.componentInstance;
+        const locationColGroup = ci.locationColGroup;
+        const genInfoColGroup = ci.genInfoColGroup;
+        const headerTemplate = ci.dynamicColGroupTemplate;
+
+        locationColGroup.headerTemplate = headerTemplate;
+        genInfoColGroup.headerTemplate = headerTemplate;
+        fixture.detectChanges();
+
+        const headerSpans: DebugElement[] = fixture.debugElement.queryAll(By.css('.dynamic-col-group-template'));
+        expect(headerSpans.length).toBe(2);
+        headerSpans.forEach(headerSpan => {
+            expect(headerSpan.nativeElement.textContent).toMatch('Dynamic column group template');
+        });
```

Comment:
Maybe we can check the template can be successfully removed as well?

Reply:
Done.